framework/scripts/tests/test_wazuh_clusterd.py

```diff
@@ -0,0 +1,362 @@
+            with patch.object(wazuh_clusterd.pyDaemonModule, 'delete_child_pids') as delete_child_pids_mock:
+                with patch.object(wazuh_clusterd.pyDaemonModule, 'delete_pid') as delete_pid_mock:
+                    with patch.object(wazuh_clusterd, 'original_sig_handler') as original_sig_handler_mock:
+                        wazuh_clusterd.exit_handler(9, 99)
+                        main_logger_mock.assert_has_calls([call.info('SIGNAL [(9)-(9)] received. Exit...')])
+                        delete_child_pids_mock.assert_called_once_with('wazuh-clusterd', 1001, main_logger_mock)
+                        delete_pid_mock.assert_called_once_with('wazuh-clusterd', 1001)
+                        original_sig_handler_mock.assert_called_once_with(9, 99)
+                        main_logger_mock.reset_mock()
+                        delete_child_pids_mock.reset_mock()
+                        delete_pid_mock.reset_mock()
+                        original_sig_handler_mock.reset_mock()
+
+                        wazuh_clusterd.original_sig_handler = original_sig_handler_not_callable
+                        wazuh_clusterd.exit_handler(9, 99)
+                        main_logger_mock.assert_has_calls([call.info('SIGNAL [(9)-(9)] received. Exit...')])
+                        delete_child_pids_mock.assert_called_once_with('wazuh-clusterd', 1001, main_logger_mock)
+                        delete_pid_mock.assert_called_once_with('wazuh-clusterd', 1001)
+                        original_sig_handler_mock.assert_not_called()
+
+
+@pytest.mark.asyncio
+async def test_master_main():
+    """Check and set the behavior of master_main function."""
+    import wazuh.core.cluster.utils as cluster_utils
+
+    class Arguments:
+        def __init__(self, performance_test, concurrency_test, ssl):
+            self.performance_test = performance_test
+            self.concurrency_test = concurrency_test
+            self.ssl = ssl
+
+    class TaskPoolMock:
+        def __init__(self):
+            self._max_workers = 1
+
+        def map(self, first, second):
+            assert first == cluster_utils.process_spawn_sleep
+            assert second == range(0, 1)
+
+    class MasterMock:
+        def __init__(self, performance_test, concurrency_test, configuration, enable_ssl, logger, cluster_items):
+            assert performance_test == 'test_performance'
+            assert concurrency_test == 'concurrency_test'
+            assert configuration == {'test': 'config'}
+            assert enable_ssl == True
+            assert logger == 'test_logger'
+            assert cluster_items == {'node': 'item'}
+            self.task_pool = TaskPoolMock()
+
+        def start(self):
+            return 'MASTER_START'
+
+    class LocalServerMasterMock:
+        def __init__(self, performance_test, logger, concurrency_test, node, configuration, enable_ssl, cluster_items):
+            pass
```

Comment:
I think here we could check the method parameters, in the same way, is done in the `MasterMock.__init__` method.